File: akka/src/main/java/com/github/uryyyyyyy/samples/akka/java/helloworld/pingpong/ParentActor.java
package com.github.uryyyyyyy.samples.akka.java.helloworld.pingpong;

import akka.actor.ActorRef;
import akka.actor.Props;
import akka.actor.UntypedActor;

public class ParentActor extends UntypedActor {

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof String) {
			System.out.println("ping received. " + message);
		} else if (message instanceof Long) {
			System.out.println("start ping. " + message);
			ActorRef echoActorRef = getContext().actorOf(Props.create(ChildActor.class), "childActor");
			echoActorRef.tell("mm", getSelf());
		} else {
			System.out.println("Got unknown type message.");
			unhandled(message);
		}
	}
}

File: akka/src/main/scala/com/github/uryyyyyyy/samples/akka/helloworld/pingpong/ParentActor.scala
package com.github.uryyyyyyy.samples.akka.helloworld.pingpong

import akka.actor.{Actor, Props}
import MyMessage.{ExecuteMessage, ReturnMessage}
import com.github.uryyyyyyy.samples.akka.helloworld.pingpong.MyMessage.ExecuteMessage

class ParentActor extends Actor {
	def receive = {
		case ExecuteMessage => {
			println("ParentActor - execute 100 * 2 and print")
			val actor = context.actorOf(Props[ChildActor])
			actor.tell(100, self)
		}
		case ReturnMessage(s) => {
			println(s"ParentActor - result is $s")
		}
	}
}